content/blog/2022/10/ContentBasedRoutingWithCamelDroolsQuarkusKogitoAltasMapApacheKafka/index.md

```diff
@@ -0,0 +1,99 @@
+
+![Using AltasMap in combination with a DMN decision table](./image-2-1536x746.png)
+
+In a [separate post](https://blog.kie.org/2022/01/data-enrichment-use-case-with-dmn-and-bpmn.html) about Data Enrichment, I previously hinted at the potentials in combining the capabilities of AltasMap with Drools DMN; I hope this tutorial now provides a very pragmatic example!
+
+## Technical details
+
+In this section, I want to highlight how the [Camel DSL](https://camel.apache.org/manual/dsl.html) allows to implement the EIP pattern very easily:
+
+```java
+from("direct:hl7")
+    .enrich("direct:label", aggregationStrategy)
+    .to("log:org.drools.demo?level=DEBUG&showAll=true&multiline=true")
+    .routingSlip(header("whereTo"))
+    .transform(HL7.ack())
+    ;
+
+from("direct:label")
+    .unmarshal().hl7()
+    .to("atlasmap:atlasmap-mapping.adm").unmarshal().json()
+    .process(kogitoDMNEvaluate) // <== Rules as DMN decisions 
+    .setHeader("topicsHeader", simple("${body[topic names]}"))
+    ;
+```
+
+As you can see, that's all we needed in order to implement the Enterprise Integration Pattern in a Quarkus application, and integrate it with AltasMap and Kogito.
+
+You can access the source code of this demo in this git repository: [https://github.com/tarilabs/quarkus-content-based-routing](https://github.com/tarilabs/quarkus-content-based-routing).
+
+## Deployment
+
+For deployment demonstration purposes:
+
+- I will make use of a Managed Service offering for Kafka, which you can try for free yourself too, by using this link: [https://red.ht/trykafka](https://red.ht/trykafka).
+- I will deploy the complete solution on the Red Hat Developer OpenShift Sandbox; you can try for free yourself by using this link: [https://developers.redhat.com/developer-sandbox](https://developers.redhat.com/developer-sandbox).
+
+After setting up the Managed Kafka and OpenShift Sandbox accounts using the links provided above, the deployment is pretty straightforward.
+
+First, we create the intended Kafka topics on the Managed Kafka console.
+
+![Creating the topic (queues) in the Managed Kafka](./image-3.png)
+
+Second, we deploy the content based routing Quarkus application using the OpenShift console.
+
+![The content based routing application now deployed on OpenShift](./image-4.png)
+
+Don't forget you can easily recreate the same setup yourself and for free, by using the links above.
+For instance, I used the very same links myself in order to make sure the demo worked fine using free resources only.
+
+Finally, the deployment and setup is complete, and we can start to make use of our content based routing solution, by sending REST calls to the ingress endpoint; this can be used as a classic webhook or analogous to a [CDS hook](https://cds-hooks.org/).
+
+![Invoking the REST API with an EDI message payload in HL7v2 format, and it is routed to the correct queue](./image-5.png)
```

Comment:
@oscerd I don;t understand if you want me to remove just the images, or the whole Deployment chapter, please?

```suggestion
```